tmt/result.py

```diff
@@ -104,70 +88,86 @@ class Result(tmt.utils.SerializableContainer):
         default_factory=list,
         serialize=lambda logs: [str(log) for log in logs],
         unserialize=lambda value: [Path(log) for log in value])
+    guest: ResultGuestData = field(
+        default_factory=ResultGuestData,
+        serialize=lambda value: value.to_serialized(),  # type: ignore[attr-defined]
+        unserialize=lambda serialized: ResultGuestData.from_serialized(serialized)
+        )
 
-    def __init__(
-            self,
+    @classmethod
+    def from_test(
+            cls,
             *,
-            data: ResultData,
-            name: Optional[str] = None,
-            test: Optional['tmt.base.Test'] = None) -> None:
-        """ Initialize test result data """
-
+            test: 'tmt.base.Test',
+            result: ResultOutcome,
+            note: Optional[str] = None,
+            duration: Optional[str] = None,
+            ids: Optional[Dict[str, Optional[str]]] = None,
+            log: Optional[List[Path]] = None,
+            guest: Optional[ResultGuestData] = None) -> 'Result':
         from tmt.base import Test
-        super().__init__()
 
-        # Save the test name and optional note
-        if not test and not name:
-            raise tmt.utils.SpecificationError(
-                "Either name or test have to be specified")
-        if test and not isinstance(test, Test):
+        if not isinstance(test, Test):
             raise tmt.utils.SpecificationError(f"Invalid test '{test}'.")
-        if name and not isinstance(name, str):
-            raise tmt.utils.SpecificationError(f"Invalid test name '{name}'.")
-
-        # ignore[union-attr]: either `name` or `test` is set, we just
-        # made sure of it above, but mypy won't realize that.
-        self.name = name or test.name  # type: ignore[union-attr]
-        self.note = data.note
-        self.duration = data.duration
-        if test:
-            # Saving identifiable information for each test case so we can match them
-            # to Polarion/Nitrate/other cases and report run results there
-            # TODO: would an exception be better? Can test.id be None?
-            self.ids = {tmt.identifier.ID_KEY: test.id}
-            for key in EXTRA_RESULT_IDENTIFICATION_KEYS:
-                self.ids[key] = test.node.get(key)
-            interpret = ResultInterpret(test.result) if test.result else ResultInterpret.RESPECT
+
+        # Saving identifiable information for each test case so we can match them
+        # to Polarion/Nitrate/other cases and report run results there
+        # TODO: would an exception be better? Can test.id be None?
```

Comment:
an unresolved `TODO` item

Reply:
Not a new one, though, just moved between the lines. Definitely out of the scope of this PR.

Reply:
Agreed.